appl/include/obj_parser.h

```diff
@@ -1,181 +1,570 @@
+/// @param uv Strcture containig the x, y component of the uv to be added
+/// @return An error number, 0 if everything have been completed succesfully
+int __obj_add_uv(obj_info_t* obj_info, const obj_float2_t uv);
+
+/// @brief Function to add a normal to the dynamic array
+/// @param obj_info Structure containig all the obj info
+/// @param normal Strcture containig the x, y, z component of the normal to be added
+/// @return An error number, 0 if everything have been completed succesfully
+int __obj_add_normal(obj_info_t* obj_info, const obj_float3_t normal);
+
+/// @brief Function to add a vertex to the dynamic array
+/// @param obj_info Structure containig all the obj info
+/// @param vertex Structure to represent a vertex with position, normal, and texture coordinates to be added
+/// @return An error number, 0 if everything have been completed succesfully
+int __obj_add_vertex(obj_info_t* obj_info, const obj_vertex_t vertex);
+
+/// @brief Function to save position information from a buffer into the obj_info structure
+/// @param obj_info Structure containig all the obj info
+/// @param buffer A buffer containing chars representing the obj file
+/// @return An error number, 0 if everything have been completed succesfully
+int __obj_save_position_info(obj_info_t* obj_info, char* buffer);
+
+/// @brief Function to save uv information from a buffer into the obj_info structure
+/// @param obj_info Structure containig all the obj info
+/// @param buffer A buffer containing chars representing the obj file
+/// @return An error number, 0 if everything have been completed succesfully
+int __obj_save_uv_info(obj_info_t* obj_info, char* buffer);
+
+/// @brief Function to save normal information from a buffer into the obj_info structure
+/// @param obj_info Structure containig all the obj info
+/// @param buffer A buffer containing chars representing the obj file
+/// @return An error number, 0 if everything have been completed succesfully
+int __obj_save_normal_info(obj_info_t* obj_info, char* buffer);
+
+/// @brief Function to save vertex information from a buffer into the obj_info structure
+/// @param obj_info Structure containig all the obj info
+/// @param buffer A buffer containing chars representing the obj file
+/// @return An error number, 0 if everything have been completed succesfully
+int __obj_save_vertex_info(obj_info_t* obj_info, char* buffer);
+
+/// @brief Function to create a new obj_info and initialize its members
+/// @return A pointer to the obj_info structure
+obj_info_t* __obj_info_new();
+
+/// @brief Function to free the memory allocated for an obj_info
+/// @param obj_info The obj_info structure to free
+void __obj_info_free(obj_info_t* obj_info);
+
+/// @brief Function to create a new obj and initialize its members
+/// @return A pointer to the obj structure
+obj_t* __obj_new();
+
+/// @brief Function to free the memory allocated for an obj
+/// @param obj The obj structure to free
+void __obj_free(obj_t* obj);
```

Comment:
obj_free must be available for users in order to destroy an obj created with obj_parse function.

Reply:
Ops...